core/src/main/java/org/apache/iceberg/hadoop/HadoopTableOperations.java

```diff
@@ -280,7 +302,7 @@ private void writeVersionHint(int versionToWrite) {
   }
 
   @VisibleForTesting
-  int readVersionHint() {
+  int versionHint() {
```

Comment:
I think a better name for this is `findVersion` because it will do listing now. It isn't really a hint because it handles the case where the hint is missing.

Reply:
Done